docs/adr/0012-clusterctl-provider.md

```diff
@@ -0,0 +1,55 @@
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+- [12. Clusterctl Config resource](#12-clusterctl-config-resource)
+  - [Context](#context)
+  - [Decision](#decision)
+  - [Consequences](#consequences)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+# 12. Clusterctl Config resource
+
+* Status: proposed
+* Date: 2024-09-03
+* Authors: @Danil-Grigorev
+* Deciders: @alexander-demicev @furkatgofurov7 @salasberryfin @Danil-Grigorev @mjura @yiannistri
+
+## Context
+
+The current implementation of clusterctl overrides is managed through `ConfigMap` managed by turtles helm chart, which has limitations when it comes to custom providers introduction and maintenance , as it requires an update for the `ConfigMap` content, followed by `rancher/turtles` release.
+
+The content of the `ConfigMap` is not easy to override without introducing issues, and it also makes the cluster configuration drift from the one installed with the `turtles` helm release.
+
+## Decision
+
+The proposed solution involves introducing a new singleton namespaced `CustomResource` - `ClusterctlConfig`, which will manage clusterctl overrides in a more flexible and maintainable way. The `ClusterctlConfig` CRD will provide `images` override options for `airGapped` installations and enable users to override or extend the default configuration with new `providers` specific to their use-case.
+
+This additional customization layer would allow to test new provider integrations early, and will not require immediate changes to `turtles` repository, which will allign it with the current [certification process](https://turtles.docs.rancher.com/tasks/provider-certification/process).
```

Comment:
```suggestion
This additional customization layer would allow to test new provider integrations early, and will not require immediate changes to `turtles` repository, which will align it with the current [certification process](https://turtles.docs.rancher.com/tasks/provider-certification/process).
```